packages/block-editor/src/utils/dom.js

```diff
@@ -162,15 +162,14 @@ export function getVisibleElementBounds( element ) {
 	let currentElement;
 
 	while ( ( currentElement = stack.pop() ) ) {
-		for ( const child of currentElement.children ) {
-			if ( isElementVisible( child ) ) {
-				let childBounds = child.getBoundingClientRect();
-				// If the parent is scrollable, use parent's scrollable bounds.
-				if ( isScrollable( currentElement ) ) {
-					childBounds = currentElement.getBoundingClientRect();
+		// Children won’t affect bounds unless the element is not scrollable.
+		if ( ! isScrollable( currentElement ) ) {
```

Comment:
I just gave this a test run. Good spotting! 

Here's this branch testing the scenarios from previous PRs:

https://github.com/user-attachments/assets/0740754c-95b0-4609-b10b-84c8e05c8dc1


https://github.com/user-attachments/assets/9151b640-7ce5-4d6c-8872-7732f8f0c12d